tools/pipeline/print-invalidations/Main.cpp

```diff
@@ -0,0 +1,67 @@
+/// \file Main.cpp
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <system_error>
+
+#include "revng/Pipeline/ContainerSet.h"
+#include "revng/Pipes/ModelGlobal.h"
+#include "revng/Pipes/PipelineManager.h"
+#include "revng/Pipes/ToolCLOptions.h"
+#include "revng/Support/CommandLine.h"
+#include "revng/Support/InitRevng.h"
+
+using std::string;
+using namespace llvm;
+using namespace llvm::cl;
+using namespace pipeline;
+using namespace ::revng::pipes;
+using namespace revng;
+
+static cl::opt<std::string> DiffPath(cl::Positional,
+                                     cl::cat(MainCategory),
+                                     cl::desc("<model diff>"),
+                                     cl::init("-"),
+                                     cl::value_desc("model"));
+
+static ToolCLOptions BaseOptions(MainCategory);
+
+static ExitOnError AbortOnError;
+
+int main(int argc, const char *argv[]) {
+  using BinaryRef = TupleTreeGlobal<model::Binary>;
+  using DiffRef = TupleTreeDiff<model::Binary>;
+
+  revng::InitRevng X(argc, argv);
+
+  HideUnrelatedOptions(MainCategory);
+  ParseCommandLineOptions(argc, argv);
+
+  Registry::runAllInitializationRoutines();
+
+  auto Manager = AbortOnError(BaseOptions.makeManager());
+  const auto &Ctx = Manager.context();
+  auto OriginalModel = *AbortOnError(Ctx.getGlobal<BinaryRef>(ModelGlobalName));
+
+  const auto &Name = ModelGlobalName;
+  auto AfterModel = AbortOnError(Ctx.getGlobal<BinaryRef>(Name));
+
+  auto Diff = AbortOnError(deserializeFileOrSTDIN<DiffRef>(DiffPath));
+  revng::ErrorList EL;
+  Diff.apply(AfterModel->get(), EL);
+  cantFail(std::move(EL));
+
+  TargetsList OverestimatedTargets;
```

Comment:
Why "overestimated"?

Reply:
because kinds do not know what is the content of containers, so they overexistamate the ones that will really be killed, and provide instead all that may be killed.